pyproject.toml

```diff
@@ -70,7 +70,7 @@ amazon = [
     "apache-airflow-providers-amazon[s3fs]>=3.0.0",
 ]
 google = ["apache-airflow-providers-google>=10.17.0"]
-microsoft = ["apache-airflow-providers-microsoft-azure"]
+microsoft = ["apache-airflow-providers-microsoft-azure>=8.5.0"]
```

Comment:
@pankajkoti would it make sense to try to fix the Airflow constraints files as well?

Reply:
I believe those constraint files are generated at the time of release and aren't mean to be mutated in order to generate reproducible builds with those constraints. The constraint files ensure that Airflow tests (existing at that time) run fine with those dependencies and the docs mention that we could upgrade/downgrade dependencies as needed. I am afraid we can fix the constraints files at this point of time.

Ref: https://airflow.apache.org/docs/apache-airflow/stable/installation/installing-from-pypi.html#reproducible-airflow-installation

Reply:
any thoughts/ideas on this @pankajastro based on your previous experience working with constraints files?

Reply:
Thanks a lot for the explanation, @pankajkoti, it makes sense!

It's a shame - I initially hoped we could leverage the Airflow constraints as they are to solve Cosmos dependencies - but I did realise this didn't quite work on a previous PR.